### PR TITLE
Improved Heap_Sort

### PR DESCRIPTION
**Pull request by imSanko**
### Describe your change:

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #9258".

### CODE REVIEW

**Comment by imSanko** [comment]
@cclauss The Checks have passed, please check PR of issue > #9528

**Comment by imSanko** [comment]
> @cclauss The Checks have passed, please check PR of issue > #9528

Kindly Can you check once

File: sorts/heap_sort.py [hunk 1]
```diff
@@ -1,58 +1,56 @@
...
-python heap_sort.py
-"""
+def heapify(arr, root, size):
+    """
+    Heapify a subtree rooted at 'root' to maintain the max-heap property.
 
+    Args:
+        arr (list): The input list.
+        root (int): The index of the root element of the subtree.
+        size (int): The size of the heap.
```
Comment (cclauss) [inline sorts/heap_sort.py:8]: As I have typed before `arr` is a weak variable name while `unsorted` is self-documenting variable name.

Please put the datatypes as typehints in the function signature where they can be verified by mypy.

Do not repeat the datatypes here because readers will get confused if one changes but the other does not.
```suggestion
        unsorted: The input list.
        root: The index of the root element of the subtree.
        size: The size of the heap.
```

File: sorts/heap_sort.py [hunk 2]
```diff
@@ -1,58 +1,56 @@
...
 
 
-def heap_sort(unsorted):
+def heap_sort(arr):
     """
-    Pure implementation of the heap sort algorithm in Python
-    :param collection: some mutable ordered collection with heterogeneous
-    comparable items inside
-    :return: the same collection ordered by ascending
+    Sorts an input list using the heap sort algorithm.
```
Comment (cclauss) [inline sorts/heap_sort.py:27]: I would revert this change.  The existing comment provides much more useful information than the proposed comment.

File: sorts/heap_sort.py [hunk 3]
```diff
@@ -1,58 +1,56 @@
...
 
-    if largest != index:
-        unsorted[largest], unsorted[index] = unsorted[index], unsorted[largest]
-        heapify(unsorted, largest, heap_size)
+    if max_index != root:
+        arr[root], arr[max_index] = arr[max_index], arr[root]
+        heapify(arr, max_index, size)
 
 
-def heap_sort(unsorted):
```
Comment (cclauss) [inline sorts/heap_sort.py:29]: This function signature needs typehints.

File: sorts/heap_sort.py [hunk 4]
```diff
@@ -1,58 +1,56 @@
...
-    :param collection: some mutable ordered collection with heterogeneous
-    comparable items inside
-    :return: the same collection ordered by ascending
+    Sorts an input list using the heap sort algorithm.
 
-    Examples:
-    >>> heap_sort([0, 5, 3, 2, 2])
-    [0, 2, 2, 3, 5]
+    Args:
+        arr (list): The input list to be sorted.
```
Comment (cclauss) [inline sorts/heap_sort.py:30]: I would revert this change.  The current doctests assure that the function does what is says it will do and provides the reader with a working (and tested) example of what the function will do with real data.  The proposed comment adds no useful information for a sorting function.

File: sorts/heap_sort.py [hunk 5]
```diff
@@ -1,58 +1,56 @@
...
 
-    >>> heap_sort([])
-    []
-
-    >>> heap_sort([-2, -5, -45])
-    [-45, -5, -2]
```
Comment (cclauss) [inline sorts/heap_sort.py:44]: Removing tests is uncool.

File: sorts/heap_sort.py [hunk 6]
```diff
@@ -1,58 +1,56 @@
...
-    user_input = input("Enter numbers separated by a comma:\n").strip()
-    unsorted = [int(item) for item in user_input.split(",")]
-    print(heap_sort(unsorted))
+    user_input = input("Enter numbers separated by commas: ").strip()
+    try:
+        unsorted = [int(item) for item in user_input.split(",")]
+        sorted_list = heap_sort(unsorted)
+        print("Sorted list:", sorted_list)
+    except ValueError:
+        print("Invalid input. Please enter a list of integers separated by commas.")
```
Comment (cclauss) [inline sorts/heap_sort.py:56]:
```suggestion
    try:
        unsorted = [int(item) for item in user_input.split(",")]
    except ValueError:
        print("Invalid input. Please enter a list of integers separated by commas.")
        exit()
    print(f"Sorted list: {heap_sort(unsorted)}")
```

**Review by cclauss — COMMENTED** [review]
Removing tests and not adding type hints is not an improvement.
